src/dev/flang/ast/AbstractType.java

```diff
@@ -1680,6 +1689,38 @@ AbstractType clone(AbstractFeature originalOuterFeature)
     return this;
   }
 
+
+  /**
+   * traverse a resolved type collecting all features this type uses.
+   *
+   * @param s the features that have already been found
+   */
+  protected abstract void usedFeatures(Set<AbstractFeature> s);
+
+
+  /**
+   * @param v
+   *
+   * @return this type and any of its generics that have more restrictive visibility than `v`.
+   */
+  public Set<AbstractFeature> moreRestrictiveVisibility(Visi v)
```

Comment:
This is ok for now, but think it might be better to use a visitor similar to `applyToGenericsAndOuter` eventually.